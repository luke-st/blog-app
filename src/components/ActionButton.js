import React from 'react'
import Fab from '@material-ui/core/Fab'
import CreateIcon from '@material-ui/icons/Create';

export const ActionButton = (props, classes) => (
    <div className='fab-container'>
        <Fab color="primary" aria-label="add" className={classes.margin}>
            <CreateIcon fontSize={'large'} />
        </Fab>
    </div>
)

export default ActionButton