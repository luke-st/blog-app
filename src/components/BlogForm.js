import React from 'react'
import moment from 'moment'
import { history } from '../routers/AppRouter'

export default class BlogForm extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            title: props.entry ? props.entry.title : '',
            body: props.entry ? props.entry.body : '',
            subtitle: props.entry ? props.entry.subtitle : '',
            createdAt: props.entry ? moment(props.entry.createdAt) : moment(),
            error: ''
        } 
    }
    onTitleChange = (e) => {
        const title = e.target.value
        this.setState(() => ({ title }))
    }
    onBodyChange = (e) => {
        const body = e.target.value
        this.setState(() => ({ body }))
    }
    onSubtitleChange = (e) => {
        const subtitle = e.target.value
        this.setState(() => ({ subtitle }))
    }
    onSubmit = (e) => {
        e.preventDefault()
        const uid = this.props.uid
        if (!this.state.title || !this.state.body) {
            this.setState(() => ({ error: 'Please provide a title and body for your post.'}))
        } else {
            this.setState(() => ({ error: ''}))
            const post = {
                title: this.state.title,
                subtitle: this.state.subtitle,
                createdAt: this.state.createdAt.valueOf(),
                body: this.state.body
            }
            this.props.onSubmit(uid, post)
            history.push('/')
        }
    }
    render () {
        return (
            <form className='form' onSubmit={this.onSubmit}>
            {this.state.error && <p className='form__error'>{this.state.error}</p>}
                <input
                    className='text-input'
                    type='text'
                    placeholder='Title'
                    autoFocus
                    value={this.state.title}
                    onChange={this.onTitleChange}
                />
                <input
                    className='text-input'
                    type='text'
                    placeholder='Subtitle'
                    autoFocus
                    value={this.state.subtitle}
                    onChange={this.onSubtitleChange}
                />
                <textarea
                    className='textarea'
                    placeholder='Write your post here!'
                    value={this.state.body}
                    onChange={this.onBodyChange}
                >
                </textarea>
                <div>
                <button className='button'>Save entry</button>
                </div>
            </form>
        )
    }
}