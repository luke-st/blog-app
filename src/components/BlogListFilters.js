import React from 'react'
import { getBloggerEntries, resetEntries } from '../actions/entries'
import { connect } from 'react-redux'
import { setTextFilter, setEndDate, setStartDate, setUID } from '../actions/filters'
import { DateRangePicker } from 'react-dates'
import 'react-dates/initialize'

class BlogListFilters extends React.Component {
    state = {
        calendarFocused: null,
        activeBlogger: this.props.auth.uid ? this.props.auth.uid : null,
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    onVloggerChange = (e) => {
        const uid = e.target.value
        this.setState({ activeBlogger:uid })
        this.props.getBloggerEntries(uid)
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    componentDidMount = () => {
        this.props.getBloggerEntries(this.state.activeBlogger)
    }
    componentWillUnmount = () => {
        this.props.resetEntries()
    }
    render() {
        return (
            <div>
                <div className='input-group'>
                    <div className='input-group__item'>
                        <select className='select'
                            value={this.state.activeBlogger}
                            onChange={this.onVloggerChange}
                        >
                        <option default>Choose a blogger</option>
                        {this.props.bloggers.map((blogger) => (
                            <option key={blogger.uid} value={blogger.uid}>{`${blogger.name}`}</option>
                        ))}
                        </select>
                    </div>
                    <div className='input-group__item'>
                        <input className='text-input' type='text' 
                                value={this.props.filters.text} 
                                onChange={this.onTextChange}
                                placeholder='Search expenses'
                            />
                    </div>
                    <div className='input-group__item'>
                            <DateRangePicker
                                startDate={this.props.filters.startDate}
                                startDateId='startDate'
                                endDate={this.props.filters.endDate}
                                endDateId='endDate'
                                onDatesChange={this.onDatesChange}
                                focusedInput={this.state.calendarFocused}
                                onFocusChange={this.onFocusChange}
                                showClearDates={true}
                                numberOfMonths={1}
                                isOutsideRange={() => false}
                            />
                        </div>
                </div>
                <div className='page-header__title'>
                {this.props.bloggers.map((blogger) => blogger.uid === this.state.activeBlogger ? (
                    <div key={blogger.uid}>
                    <p><span>{blogger.name.split(' ')[0]}</span> has <span>{blogger.posts}</span> post{blogger.posts === 1 ? '' : 's'}</p>
                    </div>
                ) : null
                )}
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters,
    bloggers: state.bloggers,
    entries: state.entries,
    auth: state.auth
})

const mapDispatchtoProps = (dispatch) => ({
    getBloggerEntries: (uid) => dispatch(getBloggerEntries(uid)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    resetEntries: () => dispatch(resetEntries())
})

export default connect(mapStateToProps, mapDispatchtoProps)(BlogListFilters)