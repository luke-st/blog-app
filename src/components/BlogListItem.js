import React from 'react'

const BlogListItem = (props) => (
    <div>
        <div className='cell'>
        <span>{props.title}</span>
        <p>{props.subtitle}</p>
        </div>
    </div>
);

export default BlogListItem