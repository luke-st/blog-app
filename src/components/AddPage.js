import React from 'react'
import { connect } from 'react-redux'
import BlogForm from './BlogForm'
import { addEntry } from '../actions/entry'

const AddPage = (props) => (
    <div className='content-container'>
        <BlogForm onSubmit={props.addEntry} uid={props.auth.uid}/>
    </div>
);

const mapStateToProps = (state) => {
    return {
        auth: state.auth 
    }
}

const mapDispatchToProps = (dispatch) => ({
    addEntry: (uid, entry) => dispatch(addEntry(uid, entry))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddPage)