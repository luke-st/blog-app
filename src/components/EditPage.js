import React from 'react'
import { connect } from 'react-redux'
import BlogForm from './BlogForm'
import { editEntry, removeEntry } from '../actions/entry'
import { history } from '../routers/AppRouter'

const EditPage = (props) => (
    <div className='content-container'>
        <BlogForm onSubmit={props.editEntry} uid={props.auth.uid} entry={props.entry} id={props.id} mode={'edit'} removeEntry={props.removeEntry} history={history}/>
    </div>
);

const mapStateToProps = (state, props) => {
    return {
        auth: state.auth,
        entry: state.entry,
        id: props.match.params.id
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    editEntry: (uid, entry) => dispatch(editEntry(uid, entry, props.match.params.id)),
    removeEntry: (uid, id) => dispatch(removeEntry(uid, id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditPage)