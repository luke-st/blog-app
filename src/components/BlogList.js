import React from 'react'
import Fab from '@material-ui/core/Fab'
import Icon from '@material-ui/core/icon/';
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import BlogListItem from './BlogListItem'

const BlogList = (props, classes) => (
    <div className='content-container'>
        <div className='grid'>
            {
                props.entries.length === 0 ? (
                    <div className='list-item list-item--message list-item--full-length'>
                        <span>No posts found ðŸ‘€</span>
                    </div>
                ) : (
                        props.entries.map((entry) => (
                            <BlogListItem key={entry.id} {...entry} />
                        ))
                    )
            }
        </div>
        {props.auth.uid ? (
            <Link to={`/add/${props.auth.uid}`}>
                <div className='fab-container'>
                    <Fab color="primary" aria-label="add" className={classes.margin}>
                        <Icon style={{ fontSize: '3rem' }}>create</Icon>
                    </Fab>
                </div>
            </Link>
        ) : null}
    </div>
)

const mapStateToProps = (state) => {
    return {
        auth: state.auth 
    }
}


export default connect(mapStateToProps)(BlogList)