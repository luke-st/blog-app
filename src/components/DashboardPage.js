import React from 'react'
import { connect } from 'react-redux'
import selectEntries from '../selectors/entries'
import BlogListFilters from './BlogListFilters'
import BlogList from './BlogList'
const DashboardPage = (props) => (
    <div>
        <div className='page-header'>
            <div className='content-container'>
                <BlogListFilters bloggers={props.bloggers} />
            </div>
        </div>
        <div className='content-container'>
            <BlogList entries={props.entries} />
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        entries: selectEntries(state.entries, state.filters),
        bloggers: state.bloggers,
        user: state.auth 
    }
}


export default connect(mapStateToProps)(DashboardPage)