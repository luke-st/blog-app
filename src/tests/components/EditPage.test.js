import React from 'react'
import { shallow } from 'enzyme'
import { EditPage } from '../../components/EditPage'

test('Should render EditPage correctly', () => {
    const wrapper = shallow(<EditPage auth={{uid: 'testUID'}} />)
    expect(wrapper).toMatchSnapshot()
})
