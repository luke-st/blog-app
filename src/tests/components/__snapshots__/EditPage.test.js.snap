// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditPage correctly 1`] = `
<div
  className="content-container"
>
  <BlogForm
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    mode="edit"
    uid="testUID"
  />
</div>
`;
